tasks/site.yml

```diff
@@ -137,6 +137,32 @@
     mode: '0644'
   tags: [ 'role::checkmk_server:multisite' ]
 
+- name: Wait for the site to be started
+  wait_for:
+    path: '{{ checkmk_server__site_home }}/{{ checkmk_server__multisite_config_path }}/wato/users.mk'
+    timeout: 60
+
+- name: Read local multisite users definition
+  command: 'sed "1,4d" {{ checkmk_server__site_home }}/{{ checkmk_server__multisite_config_path }}/wato/users.mk'
+  register: checkmk_server_register_users_mk
+  changed_when: False
+  always_run: True
+  tags: [ 'role::checkmk_server:multisite' ]
+
+- name: Set multisite users definition fact
+  set_fact:
+    checkmk_server_fact_local_users: '{{ checkmk_server_register_users_mk.stdout }}'
+  always_run: True
+  tags: [ 'role::checkmk_server:multisite' ]
+
+- name: Set local multisite user passwords
+  htpasswd:
+    path: '{{ checkmk_server__site_home }}/etc/htpasswd'
+    name: '{{ item }}'
+    password: '{{ checkmk_server__multisite_users[item]["password"] }}'
+  when: '"password" in checkmk_server__multisite_users[item]'
```

Comment:
I like to use brackets when there are more conditions to make sure that all
are considered instead of just the first one.
7 cze 2016 19:32 "Reto Gantenbein" <notifications@github.com> napisał(a):

> In tasks/site.yml
> <https://github.com/debops-contrib/ansible-checkmk_server/pull/7#discussion_r66116055>
> :
>
> > +  changed_when: False
> > +  always_run: True
> > +  tags: [ 'role::checkmk_server:multisite' ]
> > +
> > +- name: Set multisite users definition fact
> > +  set_fact:
> > +    checkmk_server_fact_local_users: '{{ checkmk_server_register_users_mk.stdout }}'
> > +  always_run: True
> > +  tags: [ 'role::checkmk_server:multisite' ]
> > +
> > +- name: Set local multisite user passwords
> > +  htpasswd:
> > +    path: '{{ checkmk_server__site_home }}/etc/htpasswd'
> > +    name: '{{ item }}'
> > +    password: '{{ checkmk_server__multisite_users[item]["password"] }}'
> > +  when: '"password" in checkmk_server__multisite_users[item]'
>
> Hmn, a quick grep didn't reveal any strict style guide in this matter. I
> personally tend to only add brackets when it helps reading conditions with
> multiple expressions. @drybjed <https://github.com/drybjed> any
> preference?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/debops-contrib/ansible-checkmk_server/pull/7/files/4409a9e88755ccf4f94873018cef2022c3361070#r66116055>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA_t9j6s5_N5tqTt6bA8wov4eb4cOuN3ks5qJasigaJpZM4IvkWW>
> .
>